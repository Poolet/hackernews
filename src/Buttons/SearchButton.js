import React from 'react';
import PropTypes from 'prop-types';

const SubmitButton = ({ value, onChange, onSubmit, children }) =>
<form onSubmit={onSubmit}>
  {children}<input
    type="text"
    value={value}
    onChange={onChange}
  />
  <button type='submit'>
    {children}
  </button>
</form>

SubmitButton.propTypes = {
  onChange: PropTypes.func,
  children: PropTypes.node,
  value: PropTypes.string,
  onSubmit: PropTypes.func
}

export default SubmitButton