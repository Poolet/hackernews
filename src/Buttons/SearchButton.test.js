import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import SearchButton from './SearchButton';
import Enzyme, { shallow, mount, render } from 'enzyme';

describe('SearchButton', () => {
  const mockSubmit = jest.fn();
  const mockChange = jest.fn();
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SearchButton onSubmit={mockSubmit}>Search</SearchButton>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('has a valid snapshot', () => {
    const component = renderer.create(
      <SearchButton onSubmit={mockSubmit}>Search</SearchButton>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls the onSubmit method when submitted', () => {
    const component = shallow(<SearchButton onSubmit={mockSubmit}>Search</SearchButton>);
    component.find('form').simulate('submit');
    expect(mockSubmit).toHaveBeenCalledTimes(1);
  });

  it('has one form', () => {
    const component = shallow(<SearchButton onSubmit={mockSubmit}>Search</SearchButton>);
    expect(component.find('form').length).toEqual(1);
  });

  it('has one button', () => {
    const component = shallow(<SearchButton onSubmit={mockSubmit}>Search</SearchButton>);
    expect(component.find('input').length).toEqual(1);
  });

  it('calls the onChange method when changed', () => {
    const component = mount(<SearchButton onChange={mockChange}>Search</SearchButton>);
    component.find('input').simulate('change');
    expect(mockChange).toHaveBeenCalledTimes(1);
  })
});
