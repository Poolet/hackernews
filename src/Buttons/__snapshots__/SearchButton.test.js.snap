// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton has a valid snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  Search
  <input
    onChange={undefined}
    type="text"
    value={undefined}
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
